src/main/java/gov/usgs/aqcu/builder/MinMaxBuilderService.java

```
@@ -11,51 +11,59 @@
-
-                // Check for Max
-                if(maxValue == null || pointValue.compareTo(maxValue) > 0) {
-                    maxValue = pointValue;
-                    maxSet = new ArrayList<>();
-                    maxSet.add(point);
-                } else if(pointValue.compareTo(maxValue) == 0) {
-                    maxSet.add(point);
-                }
-
-                // Check for Min
-                if(minValue == null || pointValue.compareTo(minValue) < 0) {
-                    minValue = pointValue;
-                    minSet = new ArrayList<>();
-                    minSet.add(point);
-                } else if(pointValue.compareTo(minValue) == 0) {
-                    minSet.add(point);
-                }
-            }
+        
+        try {
+	        if(points != null && !points.isEmpty()) {
+	            for(TimeSeriesPoint point : points) {
+	                BigDecimal pointValue = DoubleWithDisplayUtil.getRoundedValue(point.getValue());
+	
+	                // Check for Max
+	                if(maxValue == null || pointValue.compareTo(maxValue) > 0) {
+	                    maxValue = pointValue;
+	                    maxSet = new ArrayList<>();
+	                    maxSet.add(point);
+	                } else if(pointValue.compareTo(maxValue) == 0) {
+	                    maxSet.add(point);
+	                }
+	
+	                // Check for Min
+	                if(minValue == null || pointValue.compareTo(minValue) < 0) {
+	                    minValue = pointValue;
+	                    minSet = new ArrayList<>();
+	                    minSet.add(point);
+	                } else if(pointValue.compareTo(minValue) == 0) {
+	                    minSet.add(point);
+	                }
+	            }
+	        }
+	
+	        result.setMaxPoints(maxSet);
+	        result.setMinPoints(minSet);
+        } catch (Exception e) {
+        	log.error("Exception in findMinMaxPoints: ", e.getMessage());
```

Should we re-throw the error here or is it okay for this method to return a blank result when an error occurs?

---

It's ok for nothing to be returned.